### PR TITLE
PE-2815: implements the UI for creating snapshots

### PR DESCRIPTION
**Pull request by matibat**
WIP

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/3i8m7597kbni0
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/0cb0ilob1rgs8

### CODE REVIEW

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 1]
```diff
@@ -172,6 +172,9 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
       range: _range,
     ));
 
+    // // TODO: remove me
```
Comment (thiagocarvalhodev) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart]: Do we still need to remove it?

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 2]
```diff
@@ -172,6 +172,9 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
...
+    // return Uint8List.fromList([]);
+
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:177]: Remove

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 3]
```diff
@@ -283,8 +282,11 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
 
       emit(UploadingSnapshot());
 
+      // TODO: try..catch here - emit error on failure
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:285]: Unnecessary

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 4]
```diff
@@ -283,8 +282,11 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
...
       await _arweave.postTx(params.signedTx);
 
+      // emit(SnapshotUploadFailure(errorMessage: 'errorMessage'));
+
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:289]: Already implemented

File: lib/components/create_snapshot_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                          ),
+                        ],
+                        style: Theme.of(context).textTheme.bodyText1,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+
+              /// display cost of snapshot creation, already present in the state
```
Comment (matibat) [inline lib/components/create_snapshot_dialog.dart:258]: Remove. But instead: Add "Contents [of drive]", and "[Total] Drive size"

Comment (thiagocarvalhodev) [inline lib/components/create_snapshot_dialog.dart]: Maybe need to remove this comment

File: lib/components/new_button.dart [hunk 6]
```diff
@@ -375,6 +379,25 @@ PopupMenuEntry<Function> _buildCreateManifestItem(
...
+  DriveDetailLoadSuccess state,
+  bool hasMinBalance,
+) {
+  return _buildMenuItemTile(
+    context: context,
+    isEnabled: !state.driveIsEmpty && hasMinBalance,
+    itemTitle: appLocalizationsOf(context).createSnapshot,
+    // message: !state.driveIsEmpty && !hasMinBalance
+    //     ? appLocalizationsOf(context).insufficientFundsForCreateASnapshot
+    //     : null,
```
Comment (matibat) [inline lib/components/new_button.dart:393]: Remove

File: lib/services/arweave/arweave_service.dart [hunk 7]
```diff
@@ -123,6 +123,8 @@ class ArweaveService {
   ) async* {
     String cursor = '';
 
+    // return;
+
```
Comment (matibat) [inline lib/services/arweave/arweave_service.dart:127]: Remove

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart (hunk 3 above)
Comment (matibat) [reply lib/blocs/create_snapshot/create_snapshot_cubit.dart]: 94bdbe331f0e85c92c72bff001646e3508a9e8e3

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart (hunk 4 above)
Comment (matibat) [reply lib/blocs/create_snapshot/create_snapshot_cubit.dart]: 94bdbe331f0e85c92c72bff001646e3508a9e8e3

File: lib/components/new_button.dart (hunk 6 above)
Comment (matibat) [reply lib/components/new_button.dart]: 9632d0affb2d06eee35d114c651b99ebea5e076f

File: lib/services/arweave/arweave_service.dart (hunk 7 above)
Comment (matibat) [reply lib/services/arweave/arweave_service.dart]: c14f8b0b6b98201334d8b80eb240efe5e5ce74a7

File: lib/components/create_snapshot_dialog.dart (hunk 5 above)
Comment (matibat) [reply lib/components/create_snapshot_dialog.dart:258]: Will be addressed in another ticket: https://ardrive.atlassian.net/browse/PE-2933

Comment (matibat) [reply lib/components/create_snapshot_dialog.dart]: 597a547f033cb77740f98eb8010bac7f47fb7849

File: lib/services/config/app_config.dart [hunk 8]
```diff
@@ -8,18 +8,21 @@ class AppConfig {
...
     this.defaultTurboUrl,
     this.allowedDataItemSizeForTurbo,
+    this.experimentalFeaures = false,
   });
   AppConfig copyWith({
     String? defaultArweaveGatewayUrl,
     bool? useTurbo,
     String? defaultTurboUrl,
     int? allowedDataItemSizeForTurbo,
+    bool? experimentalFeaures,
```
Comment (thiagocarvalhodev) [inline lib/services/config/app_config.dart:25]: `experimentalFeaures` is too generic. When using feature flags we must be specific about each flag. We only have one, but in the future, we can have more than one feature enabled and others disabled.

Consider changing to: `enableCreateSnapshot` or `enableQuickSync`

File: lib/services/config/config_service.dart [hunk 9]
```diff
@@ -21,12 +21,13 @@ class ConfigService {
...
+      final experimentalFeatures = localStore.getString('experimentalFeatures');
       AppConfig configFromEnv = AppConfig.fromJson(json.decode(configContent));
-      if (gatewayUrl != null) {
-        configFromEnv = configFromEnv.copyWith(
-          defaultArweaveGatewayUrl: gatewayUrl,
-        );
-      }
+      configFromEnv = configFromEnv.copyWith(
+        defaultArweaveGatewayUrl: gatewayUrl,
+        experimentalFeaures: experimentalFeatures == 'YES',
```
Comment (thiagocarvalhodev) [inline lib/services/config/config_service.dart:28]: Is there a way to use `true` or `false` here?

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart (hunk 1 above)
Comment (matibat) [reply lib/blocs/create_snapshot/create_snapshot_cubit.dart]: 94bdbe331f0e85c92c72bff001646e3508a9e8e3

File: lib/components/create_snapshot_dialog.dart (hunk 5 above)
Comment (matibat) [reply lib/components/create_snapshot_dialog.dart]: 597a547f033cb77740f98eb8010bac7f47fb7849

File: lib/services/config/app_config.dart (hunk 8 above)
Comment (matibat) [reply lib/services/config/app_config.dart:25]: Nice point. Maybe `enableQuickSyncAuthoring`

File: lib/services/config/config_service.dart (hunk 9 above)
Comment (matibat) [reply lib/services/config/config_service.dart:28]: YES! I think using boolean directly is better. I'll do!

File: lib/services/config/app_config.dart (hunk 8 above)
Comment (matibat) [reply lib/services/config/app_config.dart]: 4da961ca29f71c03e3dd5483a77a43e9f205dc02

File: lib/services/config/config_service.dart (hunk 9 above)
Comment (matibat) [reply lib/services/config/config_service.dart]: 4da961ca29f71c03e3dd5483a77a43e9f205dc02

File: lib/services/config/config_service.dart [hunk 10]
```diff
@@ -21,12 +21,14 @@ class ConfigService {
       );
 
       final gatewayUrl = localStore.getString('arweaveGatewayUrl');
+      final experimentalFeatures =
```
Comment (thiagocarvalhodev) [inline lib/services/config/config_service.dart:24]: maybe just use `enableQuickSyncAuthoring`

**Review by thiagocarvalhodev — APPROVED** [review]
Left just one comment. LGTM!

File: lib/services/config/config_service.dart (hunk 10 above)
Comment (matibat) [reply lib/services/config/config_service.dart]: ac2b52ec91baa47c9f555ac97b84f772acf267b6

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

**Pull request merged**